### PR TITLE
Minor signal fix found from https://github.com/apache/nuttx/pull/8900

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Required by https://github.com/apache/nuttx/pull/8900:

- system/cu: Remove the duplicated SIGINT definition
- testing/ostest: Define the private used signal to SIGRTMIN 
- testing/ostest: Remove the unreal used SIG_WAITCANCEL

## Impact

Minor

## Testing

sim:ostest

### CODE REVIEW

**Review by patacongo — APPROVED** [review]
These all seem basically correct to me.

There is an issue with hard-coding application IPC signals to SIGRTMIN in library coding.  Sometimes the signal definition in one library could become invalid if the library is built into the same task that already uses that signal number.  I think a configuration is needed in that case.

For CU and ostest, that is not a problem because they are standalone tasks, not libraries.

There are already many signals defined in Kconfig files that have invalid default signal numbers:

   $ find . -name Kconfig | xargs grep SIG
   ./examples/ajoystick/Kconfig:config EXAMPLES_AJOYSTICK_SIGNO
   ./examples/alarm/Kconfig:config EXAMPLES_ALARM_SIGNO
   ./examples/buttons/Kconfig:     default EXAMPLES_BUTTONS_SIGNAL
   ./examples/buttons/Kconfig:config EXAMPLES_BUTTONS_SIGNAL
   ./examples/djoystick/Kconfig:config EXAMPLES_DJOYSTICK_SIGNO
   ./examples/nx/Kconfig:config EXAMPLES_NX_NOTIFYSIGNO
   ./examples/nxhello/Kconfig:config EXAMPLES_NXHELLO_NOTIFYSIGNO
   ./examples/nximage/Kconfig:config EXAMPLES_NXIMAGE_NOTIFYSIGNO
   ./examples/nxlines/Kconfig:config EXAMPLES_NXLINES_NOTIFYSIGNO
   ./examples/nxscope/Kconfig:config EXAMPLES_NXSCOPE_TIMER_SIGNO
   ./examples/nxterm/Kconfig:config EXAMPLES_NXTERM_NOTIFYSIGNO
   ./examples/oneshot/Kconfig:config EXAMPLES_ONESHOT_SIGNO
   ./examples/timer/Kconfig:config EXAMPLES_TIMER_SIGNO
   ./examples/timer_gpio/Kconfig:config EXAMPLES_TIMER_GPIO_SIGNO
   ./examples/watcher/Kconfig:config EXAMPLES_WATCHER_SIGNAL
   ./examples/watcher/Kconfig:config EXAMPLES_WATCHER_SIGNAL_LOG
   ./examples/zerocross/Kconfig:config EXAMPLES_ZEROCROSS_SIGNO
   ./graphics/ft80x/Kconfig:config GRAPHICS_FT80X_CMDEMPTY_SIGNAL
   ./graphics/ft80x/Kconfig:config GRAPHICS_FT80X_TAG_SIGNAL
   ./graphics/nxwm/Kconfig:config NXWM_TOUCHSCREEN_SIGNO
   ./graphics/nxwm/Kconfig:config NXWM_KEYBOARD_SIGNO
   ./graphics/nxwm/Kconfig:config NXWM_CALIBRATION_SIGNO
   ./netutils/dhcpd/Kconfig:config NETUTILS_DHCPD_SIGWAKEUP
   ./netutils/netinit/Kconfig:config NETINIT_SIGNO
   ./netutils/ntpclient/Kconfig:config NETUTILS_NTPCLIENT_SIGWAKEUP

Other places in C and header files where signals are defined:

    ./examples/ajoystick/ajoy_main.c:#  define CONFIG_EXAMPLES_AJOYSTICK_SIGNO 13
    ./examples/buttons/buttons_main.c:#  define CONFIG_EXAMPLES_BUTTONS_SIGNO 13
    ./examples/chrono/chrono_main.c:#define BUTTON_SIGNO 13
    ./examples/djoystick/djoy_main.c:#  define CONFIG_EXAMPLES_DJOYSTICK_SIGNO 13
    ./examples/oneshot/oneshot_main.c:#  define CONFIG_EXAMPLES_ONESHOT_SIGNO 13
    ./examples/timer/timer_main.c:#  define CONFIG_EXAMPLES_TIMER_SIGNO 17
    ./examples/zerocross/zerocross_main.c:#  define CONFIG_EXAMPLES_ZEROCROSS_SIGNO 13
    ./testing/drivertest/drivertest_timer.c:#define TIMER_DEFAULT_SIGNO 17

    ./examples/nx/nx_internal.h:#  define CONFIG_EXAMPLES_NX_NOTIFYSIGNO 4
    ./examples/nxdemo/nxdemo.h:#  define CONFIG_EXAMPLES_NXDEMO_NOTIFYSIGNO 4
    ./examples/nxhello/nxhello.h:#  define CONFIG_EXAMPLES_NXHELLO_NOTIFYSIGNO 4
    ./examples/nximage/nximage.h:#  define CONFIG_EXAMPLES_NXIMAGE_NOTIFYSIGNO 4
    ./examples/nxlines/nxlines.h:#  define CONFIG_EXAMPLES_NXLINES_NOTIFYSIGNO 4
    ./examples/nxterm/nxterm_internal.h:#  define CONFIG_EXAMPLES_NXTERM_NOTIFYSIGNO 4
    ./games/shift/shift_input_joystick.h:#define DJOYSTICK_SIGNO 13
    ./include/netutils/netinit.h:#  define CONFIG_NETINIT_SIGNO 18
    ./include/netutils/ntpclient.h:#  define CONFIG_NETUTILS_NTPCLIENT_SIGWAKEUP 18

**Comment by pkarashchenko** [comment]
I would not state that those are "invalid". There is a code that re-use one of the standard signals. Like
```
signal(CONFIG_EXAMPLES_BUTTONS_SIGNO, SIG_IGN);

      sigemptyset(&set);
      sigaddset(&set, CONFIG_EXAMPLES_BUTTONS_SIGNO);
      ret = sigwaitinfo(&set, &value);
```
So this can work in theory, however I think that `sighold(CONFIG_EXAMPLES_BUTTONS_SIGNO)` should be added otherwise signal will bypass sigqueue

**Comment by xiaoxiang781216** [comment]
Ok, the default value change to 30 and remove the unused macros.

**Comment by patacongo** [comment]
> I would not state that those are "invalid".

I think it depends on how much of a purist you are.  Most of the standard signals have not yet been implemented so re-using an unimplemented standard signal is safe.

However, I intend to look at fixing some of the errors with real time time signal prioritization.  So if you really want to participate is prioritized real time signal behavior, you have to use the signals in the real time signal range (SIGRTMIN through SIGRTMAX).  Standard signals have different behaviors:  The no required to be queue; if queued, they should be FIFO.

**Comment by pkarashchenko** [comment]
> > I would not state that those are "invalid".
> 
> I think it depends on how much of a purist you are. Most of the standard signals have not yet been implemented so re-using an unimplemented standard signal is safe.
> 
> However, I intend to look at fixing some of the errors with real time time signal prioritization. So if you really want to participate is prioritized real time signal behavior, you have to use the signals in the real time signal range (SIGRTMIN through SIGRTMAX). Standard signals have different behaviors: The no required to be queue; if queued, they should be FIFO.
> 
> NuttX current queues all signals FIFO.

Yes. I agree for sure. So if application is interested if the event happened and does not care how many times or even producer can publish number of times in`si_value` the one of the "standard" signal number can be used. But if application is interested in every event (like all "mount"/"unmount" events) then definitely realtime signals + FIFO is the only option.
I just intended to say that not all results of the `gerp` that you provided should be blindly changed to 30.

**Comment by pkarashchenko** [comment]
Actually there are two FIFOs :) The first one is the FIFO of different signal values and a FIFO of realtime signals of a same value

**Comment by patacongo** [comment]
> I just intended to say that not all results of the `gerp` that you provided should be blindly changed to 30.

There is no emergency that requires a change in those cases.  But I don't think this is a loophole that the typical user can use:  How could they know which standard signals are safe to reuse for a different purpose.

Bad code tends to get replicated numerous time so it would be nice only to have pristine code examples in apps, but not necessary.

**Comment by pkarashchenko** [comment]
> How could they know which standard signals are safe to reuse for a different purpose.

Most likely `SIGUSR1` and `SIGUSR2` can be used for sure

**Comment by pkarashchenko** [comment]
I think we should wait for https://github.com/apache/nuttx/pull/8885 to get merged as 30 now is an RT signal, but after the PR it will be in standard signal range.

**Comment by patacongo** [comment]
In the Kconfig files, when a signal number value is selectable, it should be limited by:

    range 32 63

With nuttx PR 8885, we will have have RTSIG_MAX of 32 and (), SIGRTMIN=32 and SIGRTMIN=63 which is in agreement with Linux.

The signals 30 and 31 are reserved.  Linux does that to; it uses some hidden internal signals with each process so that the range of standard and real time signals has a small hole it too.

**Comment by patacongo** [comment]
> Most likely `SIGUSR1` and `SIGUSR2` can be used for sure. Or I'm missing something?

Correct.  They exist only because they were defined many years before the POSIX real time extensions that added real time signal support.  So they are not being re-used; that is their intended use.  They are precursor real time signals with properties like standard signals.

File: examples/alarm/Kconfig [hunk 1]
```diff
@@ -33,6 +33,6 @@ config EXAMPLES_ALARM_DEVPATH
 
 config EXAMPLES_ALARM_SIGNO
 	int "Alarm signal"
-	default 1
```
Comment (pkarashchenko) [inline examples/alarm/Kconfig:36]: Maybe we can default to `SIGRTMIN`?
or maybe add some range check to make sure that this is a realtime signal

File: examples/README.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ Example Configuration:
 - `CONFIG_EXAMPLES_AJOYSTICK_DEVNAME` – Joystick device name. Default
   `/dev/adjoy0`.
 - `CONFIG_EXAMPLES_AJOYSTICK_SIGNO` – Signal used to signal the test
-  application. Default `13`.
+  application. Default `30`.
```
Comment (pkarashchenko) [inline examples/README.md:63]: Now it need to be moved at least to SIGRTMIN (aka 32)

**Review by patacongo — APPROVED** [review]
LGTM

File: examples/alarm/Kconfig (hunk 1 above)
Comment (patacongo) [reply examples/alarm/Kconfig:36]: Can this be resolved?  This PR does not address the Kconfig signal number settings any more.

File: examples/README.md (hunk 2 above)
Comment (xiaoxiang781216) [reply examples/README.md]: Done.

File: examples/alarm/Kconfig (hunk 1 above)
Comment (xiaoxiang781216) [reply examples/alarm/Kconfig:36]: Done.

**Comment by patacongo** [comment]
Looks like this is trying to use the NuttX buildroot toolchain:

    /usr/bin/bash: arm-nuttx-eabi-gcc: command not found

**Comment by pkarashchenko** [comment]
Some configs have it enabled, but CI changes the toolchain to either GNU_EABI or CLANG, so the actual build is done with one of those

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 could you please rebase PR?

**Comment by xiaoxiang781216** [comment]
Done.

File: include/graphics/twm4nx/twm4nx_config.hxx [hunk 3]
```diff
@@ -448,7 +448,7 @@
  */
 
 #ifndef CONFIG_TWM4NX_INPUT_SIGNO
-#  define CONFIG_TWM4NX_INPUT_SIGNO 6
+#  define CONFIG_TWM4NX_INPUT_SIGNO 33
```
Comment (davids5) [inline include/graphics/twm4nx/twm4nx_config.hxx:451]: typo?

**Review by davids5 — COMMENTED** [review]
Is 33 a typo?

File: include/graphics/twm4nx/twm4nx_config.hxx (hunk 3 above)
Comment (xiaoxiang781216) [reply include/graphics/twm4nx/twm4nx_config.hxx:451]: No, this problem use two signal 5/6 at th same time, we need map to 32/33.

**Review by pkarashchenko — APPROVED** [review]

**Review by davids5 — APPROVED** [review]

**Pull request merged**